File: java/src/test/BasicUserTest.java
package mx.iteso;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;
import static org.junit.Assert.Equals;
import static org.junit.Assert.assertTrue;
import static org.junit.Before;
import static org.junit.Test;

/**
 * Basic user test class.
 */
public class BasicUserTest {
    private User user;

    /**
     * Set up method.
     */
    @Before
    public void SetUp(){
        user = mock((BasicUser.class));
    }    
    /**
     * Get name test.
     */
    @Test
    public void NameTest(){
        when(user.getName()).thenReturn("Basic User");
    }
}

File: csharp/decorator_practice/decorator_practice/test/BasicUserTest.cs
using System;
using System.Collections.Generic;
using System.Text;
using decorator_practice.main.decorator;
using decorator_practice.main.users;
using Moq;
using NUnit.Framework;

namespace decorator_practice.test
{
    [TestFixture]
    class BasicUserTest
    {
        public Mock<User> user;
        public BasicUser bu;

        [SetUp]
        public void SetUp()
        {
            bu = new BasicUser();
            user = new Mock<User>(bu);
        }

        [TestCase]
        public void NameTest()
        {
            Assert.AreEqual("Basic User", user.Object.getName());
        }
    }
}